contrib/vexecutor/execVQual.c

```diff
@@ -123,3 +123,431 @@ VirtualNodeProc(ScanState* state,TupleTableSlot *slot){
+			bool *isNull, ExprDoneCond *isDone)
+{
+	List	   *clauses = andExpr->args;
+	ListCell   *clause;
+	bool		AnyNull;
+	vbool	*res = NULL;
+	vbool	*next = 	NULL;
+	bool		skip = true;
+	int 		i = 0;
+
+	if (isDone)
+		*isDone = ExprSingleResult;
+
+	AnyNull = false;
+
+	/*
+	 * If any of the clauses is FALSE, the AND result is FALSE regardless of
+	 * the states of the rest of the clauses, so we can stop evaluating and
+	 * return FALSE immediately.  If none are FALSE and one or more is NULL,
+	 * we return NULL; otherwise we return TRUE.  This makes sense when you
+	 * interpret NULL as "don't know", using the same sort of reasoning as for
+	 * OR, above.
+	 */
+
+	foreach(clause, clauses)
+	{
+		ExprState  *clausestate = (ExprState *) lfirst(clause);
+		Datum		clause_value;
+
+		/*
+		 * to check if all the values is false, then skip to evaluate some
+		 * expressions
+		 */
+		skip = true;
+
+		clause_value = ExecEvalExpr(clausestate, econtext, isNull, NULL);
+
+		if(NULL == res)
+		{
+			res = DatumGetPointer(clause_value);
+			for(i = 0; i < res->header.dim; i++)
+			{
+				if(res->header.isnull[i] || res->values[i])
+				{
+					skip = false;
+					break;
+				}
+			}
+		}
+		else
+		{
+			next = DatumGetPointer(clause_value);
+			for(i = 0; i < res->header.dim; i++)
+			{
+				res->header.isnull[i] = (res->header.isnull[i] ||next->header.isnull[i]);
+				res->values[i] = (res->values[i] || next->values[i]);
+				if(skip && (res->header.isnull[i] || res->values[i]))
+					skip = false;
+			}
+		}
+
+		if(skip)
+		{
+			*isNull = false;
+			return PointerGetDatum(res);
+		}
+	}
+
+	*isNull = false;
+	return PointerGetDatum(res);
+}
+
+/*
+ * Init the vectorized expressions
+ */
+ExprState *
+VExecInitExpr(Expr *node, PlanState *parent)
+{
+	ExprState *state = NULL;
+	if(NULL == parent->vectorized)
+		return NULL;
+
+	/*
+	 * Because Var is the leaf node of the expression tree, it have to be
+	 * refactored first, otherwise the all call stack should be refactored.
+	 */
+	switch (nodeTag(node))
+	{
+		case T_Var:
+			state = (ExprState *) makeNode(ExprState);
+			state->evalfunc = VExecEvalVar;
+			break;
+		case T_BoolExpr:
+			{
+				BoolExpr   *boolexpr = (BoolExpr *) node;
+				BoolExprState *bstate = makeNode(BoolExprState);
+
+				switch (boolexpr->boolop)
+				{
+					case AND_EXPR:
+						bstate->xprstate.evalfunc = (ExprStateEvalFunc) VExecEvalAnd;
+						break;
+					case OR_EXPR:
+						bstate->xprstate.evalfunc = (ExprStateEvalFunc) VExecEvalOr;
+						break;
+					case NOT_EXPR:
+						bstate->xprstate.evalfunc = (ExprStateEvalFunc) VExecEvalNot;
+						break;
+					default:
+						elog(ERROR, "unrecognized boolop: %d",
+							 (int) boolexpr->boolop);
+						break;
+				}
+				bstate->args = (List *)
+					ExecInitExpr((Expr *) boolexpr->args, parent);
+				state = (ExprState *) bstate;
+			}
+			break;
+		/*TODO: More and more expressions should be vectorized */
+		default:
+			break;
```

Comment:
It's better to think about how to fallback to original executor when we find any unsupported behavior to make sure the query could have right result. There are so many places that just error out or assert(false) which is not friendly.